libs/api/domains/license-service/src/lib/client/firearm-license-client/firearmLicenseService.api.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  GenericLicenseClient,
+  GenericLicenseUserdataExternal,
+  GenericUserLicensePkPassStatus,
+  GenericUserLicenseStatus,
+  PkPassVerification,
+} from '../../licenceService.type'
+import type { Logger } from '@island.is/logging'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import {
+  FirearmApplicationApi,
+  LicenseInfo,
+} from '@island.is/clients/firearm-license'
+import { User } from '@island.is/auth-nest-tools'
+import { Inject } from '@nestjs/common'
+import { parseFirearmLicensePayload } from './firearmLicenseMapper'
+import { FetchError } from '@island.is/clients/middlewares'
+
+/** Category to attach each log message to */
+const LOG_CATEGORY = 'firearmlicense-service'
+
+export class GenericFirearmLicenseApi
+  implements GenericLicenseClient<LicenseInfo> {
+  constructor(
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+    private firearmApi: FirearmApplicationApi,
+  ) {}
+
+  private handleError(error: Partial<FetchError>): unknown {
+    // Not throwing error if service returns 403 or 404. Log information instead.
+    if (error.status === 403 || error.status === 404) {
+      this.logger.info(`Firearm license returned ${error.status}`, {
+        exception: error,
+        message: (error as Error)?.message,
+        category: LOG_CATEGORY,
+      })
+      return null
+    }
+    this.logger.info('ADR license fetch failed', {
+      exception: error,
+      message: (error as Error)?.message,
+      category: LOG_CATEGORY,
+    })
+
+    return null
+  }
+
+  async fetchLicense(user: User) {
+    let license: unknown
+
+    try {
+      license = await this.firearmApi.apiFirearmApplicationLicenseInfoSsnGet({
+        ssn: user.nationalId,
+      })
+    } catch (e) {
+      this.handleError(e)
+    }
+
+    return license as LicenseInfo
+  }
+
+  async getLicense(user: User): Promise<GenericLicenseUserdataExternal | null> {
+    const license = await this.fetchLicense(user)
+
+    if (!license) {
+      this.logger.warn('Missing Firearm license, null from api', {
+        category: LOG_CATEGORY,
+      })
```

Comment:
```suggestion
```

We handle logging in `fetchLicense` for all cases, this would just create extra noise.